project-base/storefront/components/Blocks/Product/AddToCart.tsx

```diff
@@ -59,25 +59,27 @@ export const AddToCart: FC<AddToCartProps> = ({
 
     return (
         <div className={twMergeCustom('flex items-stretch justify-between gap-2', className)}>
-            <Spinbox
-                defaultValue={1}
-                id={productUuid}
-                max={maxQuantity}
-                min={minQuantity}
-                ref={spinboxRef}
-                size="small"
-                step={1}
-            />
+            {maxQuantity !== 0 && (
+                <Spinbox
+                    defaultValue={1}
+                    id={productUuid}
+                    max={maxQuantity}
+                    min={minQuantity}
+                    ref={spinboxRef}
+                    size="small"
+                    step={1}
+                />
+            )}
             <Button
-                className="py-2"
-                isDisabled={fetching}
+                className={`py-2 ${maxQuantity === 0 ? 'w-full' : ''}`}
+                isDisabled={fetching || maxQuantity === 0}
```

Comment:
I believe the product quantity might be also negative, so I would change all these conditions to `maxQuantity <= 0`

Reply:
(even thought I was not able to set the negative value in the admin :sweat_smile: )

Reply:
ah that is for sure more complete solution, changed all other conditions as such